sim_pkg/__main__.py

```diff
@@ -6,85 +8,37 @@
 import socketserver
 import csv
 import numpy as np
-from init_pose import init
 import argparse
+from sim_pkg.config import Configuration
 
-# sim_process_ = 0 
 robot_processes_ = []
-# vis_processes_ = 0
 parser = argparse.ArgumentParser(description='Run the simulator')
 
-parser.add_argument('-fn', '--filename', type=str, help="Name of the file without the .py extension")
+parser.add_argument('-fn', '--filename', type=str,
+                    help="Name of the file without the .py extension")
 
 args = parser.parse_args()
 
-def write_to_csv(free_port):
-    """
-    Write avaliable port number to csv file
-    """
-    f = open('port.csv', 'w')
-    num = [free_port]
-    with f:
-        # create the csv writer
-        writer = csv.writer(f)
-
-        # write a row to the csv file
-        writer.writerow(num)
-
-    # close the file
-    f.close()
-
-def init_pos(swarmsize):
-    """
-    Initialize the position
-    """
-    x = [0]*swarmsize
-    y = [0]*swarmsize
-    theta = [0]*swarmsize
-    a_ids = [0]*swarmsize
-
-    x, y, theta, a_ids = init(swarmsize,x,y,theta,a_ids)
-    id = [i for i in range(swarmsize)]
-
-    data = np.array([id, x, y, theta, a_ids]).T
-    np.savetxt("init.csv", data, delimiter=",", fmt="%.2f",
-           header="ID, x, y, theta, a_ids")
-
-
 def run():
     """
     Opens all the programs involved
     """
 
-    # Finds the first available socket on the system
-    with socketserver.TCPServer(("localhost", 0), None) as s:
-        free_port = s.server_address[1]
-
-    # Writes the socket number to port.csv file 
-    write_to_csv(free_port)
-    
     # Read required parameters from config.json
+    config = Configuration.from_path('config.json')  # Shouldn't be hardcoded.
     path = str(Path(__file__).parent)
-    with open('config.json', 'r') as myfile:
-        data=myfile.read()
-    config_var = json.loads(data)
```

Comment:
I hate this. This function has a scope that is so loaded and hard to read. See my solution. In my solution `Configuration` encapsulates all the configuration values.